### ISSUE TITLE
Local environment setup failed[1]

### ISSUE THREAD

**Issue by rat-moonshine** [prominic/Moonshine-IDE#785]
Lately we are seeing lots of this message/alert pops-up during importing a project into Moonshine workspace. 

![image](https://user-images.githubusercontent.com/24370870/108477373-24644780-72b9-11eb-8d2a-0c6742e65be7.png)

Based on the Adobe documentation, Error#3013 fires when file/directory is in use.

I can imagine that writing the batch file to setup local environment in a multiple close calls - may create this file-holding issue. We already have seen file-holding problem in Windows environment. This error also particularly shown in Windows run as the batch file generates on Windows run only.

Although, the generation process runs in a queued manner, but we may need to depends more on the file-writing asynchronous process rather than synchronous way.

**Comment by joshtynjala** [comment]
In December, I made a commit to fix the Windows error "the process cannot access the file because it is being used by another process":

176dcabdf824ec69e9f8779999fc9ed07c386acd

I had to switch from using AIR's asynchronous file APIs to the synchronous version because the file was not actually fully closed when the Event.CLOSE event indicated that it was. Even after a delay of a full second, it sometimes failed. It's possible that your issue is similar. The synchronous file APIs should work better, so I'd look for other places where the async APIs are still being used.

**Comment by rat-moonshine** [comment]
Unfortunately, we're still encountering the error time to time on Windows in a close calls. This particularly noticeable in a VD environment. 

Isn't this that having an asynchronous process, it is more safe? When synchronouss call in close-interval could be more write-error prone? The environment-setup util anyway run its process in a queued manner, but of course this queue depends on the return of asynchronous write process events.

It could be better if we able to run this on-fly, but because of different reasons, we found it's working mostly when written to a batch script along with other run-command at its end.

I have added another check anyway to the process, although this may not reduce the calls in a great manner - the util now check (if) the existing batch file's content is same to the requested content, and if matched it'll skip to running the file, instead of writing.

**Comment by joshtynjala** [comment]
When I changed from async to sync, it significantly reduced the number of times that I saw an error. Based on that experience, async is definitely not more safe. It's much, much, much less safe.

**Comment by rat-moonshine** [comment]
Yeah, it's a pity when an API does not provides what expects from it, correctly.

I was thinking of another probability, if the writing process is not entirely to blame in this case. If it's the holding by the complete handler's NativeProcess which runs the batch file, and another process tries to write the file at same time.

But any way, this is all related to writing/running the same file located somewhere. 

So I changed the file generation process now. I'm generating separate files on each environment-setup process to temp directory. This temp directory deletes during Moonshine exits. With this we should not have same file write/access problem anymore. 

I have also restored synchronous file writing in the class.

So far tested on Windows, I didn't noticed "Local environment setup failed[1]" error/alert anymore. 

I have updated test links in #756 ;
